src/chain/redb/v1.rs

```diff
@@ -311,11 +311,22 @@ impl ChainStore {
     }
 
     pub fn get_tx(&self, tx_hash: &[u8]) -> Result<Option<Vec<u8>>, Error> {
+        let tx_with_block = self.get_tx_with_block_data(tx_hash)?;
+        if let Some((_, tx)) = tx_with_block {
+            return Ok(Some(tx));
+        }
+        Ok(None)
+    }
+
+    pub fn get_tx_with_block_data(
+        &self,
+        tx_hash: &[u8],
+    ) -> Result<Option<(BlockBody, Vec<u8>)>, Error> {
```

Comment:
I think `BlockBody` used everywhere in this PR is the wrong name. A block is very loosely split into two things:
1. block header
2. block body

`MultiEraBlock` consists of the full block (body, header, anything else). That's also what I think is logical to return from utxorpc as well. However, in this PR, you refer to the fully block as "BlockBody" which is misleading because it makes it sound like you're returning only the "body" field of the block (whereas you're actually returning the full block)

Reply:
maybe just to loop in @scarmuega since I'm not sure if there was a separate rationale for calling things "body"